ctx.go

```diff
@@ -28,22 +28,263 @@ import (
+		Redirect(location string, status ...int)
+
+		// Render a template with data and sends a text/html response.
+		// We support the following engines: html, amber, handlebars, mustache, pug
+		Render(name string, bind interface{}, layouts ...string) (err error)
+
+		// Route returns the matched Route struct.
+		Route() *Route
+
+		// SaveFile saves any multipart file to disk.
+		SaveFile(fileheader *multipart.FileHeader, path string) error
+
+		// Secure returns a boolean property, that is true, if a TLS connection is established.
+		Secure() bool
+
+		// Send sets the HTTP response body. The input can be of any type, io.Reader is also supported.
+		Send(bodies ...interface{})
+
+		// SendBytes sets the HTTP response body for []byte types without copying it.
+		// From this point onward the body argument must not be changed.
+		SendBytes(body []byte)
+
+		// SendFile transfers the file from the given path.
+		// The file is not compressed by default, enable this by passing a 'true' argument
+		// Sets the Content-Type response HTTP header field based on the filenames extension.
+		SendFile(file string, compress ...bool) error
+
+		// SendStatus sets the HTTP status code and if the response body is empty,
+		// it sets the correct status message in the body.
+		SendStatus(status int)
+
+		// SendString sets the HTTP response body for string types.
+		// This means no type assertion, recommended for faster performance
+		SendString(body string)
+
+		// SendStream sets response body stream and optional body size.
+		SendStream(stream io.Reader, size ...int)
+
+		// Set sets the response's HTTP header field to the specified key, value.
+		Set(key string, val string)
+
+		// Subdomains returns a string slice of subdomains in the domain name of the request.
+		// The subdomain offset, which defaults to 2, is used for determining the beginning of the subdomain segments.
+		Subdomains(offset ...int) []string
+
+		// Stale is not implemented yet, pull requests are welcome!
+		Stale() bool
+
+		// Status sets the HTTP status for the response.
+		Status(status int)
+
+		// Type sets the Content-Type HTTP header to the MIME type specified by the file extension.
+		Type(extension string, charset ...string)
+
+		// Vary adds the given header field to the Vary response header.
+		// This will append the header, if not already listed, otherwise leaves it listed in the current location.
+		Vary(fields ...string)
+
+		// Write appends any input to the HTTP body response, io.Reader is also supported as input.
+		Write(bodies ...interface{})
+
+		// XHR returns a Boolean property, that is true, if the request's X-Requested-With header field is XMLHttpRequest,
+		// indicating that the request was issued by a client library (such as jQuery).
+		XHR() bool
+	}
+
+	// Context represents the Context which hold the HTTP request and response.
+	// It has methods for the request query string, parameters, body, HTTP headers and so on.
+	Context struct {
```

Comment:
What about making Context struct private?

Reply:
I made it private, however AcquireCtx() now return the unexported type of this struct.